daemon/containerd/image_delete.go

```diff
@@ -45,18 +46,25 @@ import (
 // TODO(thaJeztah): implement ImageDelete "force" options; see https://github.com/moby/moby/issues/43850
 // TODO(thaJeztah): implement ImageDelete "prune" options; see https://github.com/moby/moby/issues/43849
 // TODO(thaJeztah): add support for image delete using image (short)ID; see https://github.com/moby/moby/issues/43854
-// TODO(thaJeztah): mage delete should send image "untag" events and prometheus counters; see https://github.com/moby/moby/issues/43855
+// TODO(thaJeztah): image delete should send image "untag" events and prometheus counters; see https://github.com/moby/moby/issues/43855
 func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
 	parsedRef, err := reference.ParseNormalizedNamed(imageRef)
 	if err != nil {
 		return nil, err
 	}
 	ref := reference.TagNameOnly(parsedRef)
 
+	img, err := i.GetImage(ctx, imageRef, image.GetImageOpts{})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Ah, right, so we need to do this separately before the actual delete, because we otherwise can no longer get the image information. Would be nice if `i.client.ImageService().Delete()` would return the image / information about the image it deleted, i.e. change the signature to

```go
Delete(ctx context.Context, name string, opts ...DeleteOpt) (Image, error)
```

I had a quick look at `i.GetImage()` and it looks to be doing _a lot_, and it seems we _only_ need the `img.ID()`, which (please double check) I think is directly the digest of the descriptor;

https://github.com/moby/moby/blob/2fa8f54d0869511f88efd318f67fe1d06915e78a/daemon/containerd/image.go#L34-L39

https://github.com/moby/moby/blob/2fa8f54d0869511f88efd318f67fe1d06915e78a/daemon/containerd/image.go#L73

So perhaps we can replace this with;

```suggestion
	desc, err := i.resolveDescriptor(ctx, imageRef)
	if err != nil {
		return nil, err
	}
	imgID := string(desc.Digest)
```

Reply:
Fixed, thanks